internal/ls/findallreferences.go

```diff
@@ -466,11 +467,11 @@ func (l *LanguageService) getImplementationReferenceEntries(ctx context.Context,
 	return core.FlatMap(symbolsAndEntries, func(s *SymbolAndEntries) []*ReferenceEntry { return s.references })
 }
 
-func (l *LanguageService) ProvideRename(ctx context.Context, params *lsproto.RenameParams) (lsproto.WorkspaceEditOrNull, error) {
+func (l *LanguageService) ProvideRename(ctx context.Context, params *lsproto.RenameParams, prefs *lsutil.UserPreferences) (lsproto.WorkspaceEditOrNull, error) {
 	program, sourceFile := l.getProgramAndFile(params.TextDocument.Uri)
 	position := int(l.converters.LineAndCharacterToPosition(sourceFile, params.Position))
 	node := astnav.GetTouchingPropertyName(sourceFile, position)
-	if node.Kind != ast.KindIdentifier {
+	if node.Kind != ast.KindIdentifier && !(ast.IsStringLiteralLike(node) && tryGetImportFromModuleSpecifier(node) != nil && prefs.AllowRenameOfImportPath == core.TSTrue) {
```

Comment:
AllowRenameOfImportPath handling seems off, maybe? I'm noticing that the old code only checks it once. Maybe the structure really is that different